packages/medusa/src/services/cart.ts

```diff
@@ -1021,24 +1049,26 @@ class CartService extends TransactionBaseService {
               )
             }
 
-            const variantsPricing = await this.pricingService_
-              .withTransaction(transactionManager)
-              .getProductVariantsPricing(
-                [
+            if (should_calculate_prices) {
```

Comment:
question: shouldn't we always calculate the prices when we update the line item?

Reply:
no, this was how it was handled previously which makes sense from a store perspective, but breaks custom unit prices set from the draft orders (context: #5274) 😄 made the same mistake in the initial pr introducing this calculation

Reply:
ahah got it 👍 thanks for the context